### PR TITLE
fix(ReactImageView): Add features

### LINKED ISSUE #278: Fix remaining Image features
The following Image features are not working in UIExplorer:
- Resize Method
- Tint Color
- Background Color
- Plain Static Image

### PR DESCRIPTION
**Pull request by ebragge**
resizeMode
tintColor
backgroundColor
imageSize

Fixes #278, #366

### CODE REVIEW

File: Libraries/Image/Image.windows.js [hunk 1]
```diff
@@ -113,6 +113,21 @@ var Image = React.createClass({
 
   statics: {
     resizeMode: ImageResizeMode,
+
+    getSize(
```
Comment (rozele) [inline Libraries/Image/Image.windows.js]: While we're touching this file, can we do a full diff against Image.android.js / Image.windows.js and see if there are maybe any other JS-only features?

File: ReactWindows/ReactNative/Modules/Image/ImageLoaderModule.cs [hunk 2]
```diff
@@ -1,9 +1,42 @@
...
 {
     class ImageLoaderModule : NativeModuleBase
     {
+        private readonly Dictionary<BitmapImage, ExceptionRoutedEventHandler> _imageFailedHandlers =
+            new Dictionary<BitmapImage, ExceptionRoutedEventHandler>();
+
+        private readonly Dictionary<BitmapImage, RoutedEventHandler> _imageOpenedHandlers =
+            new Dictionary<BitmapImage, RoutedEventHandler>();
+
+        private readonly Dictionary<string, BitmapImage> _images =
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Image/ImageLoaderModule.cs]: This doesn't seem like a good idea to me to keep these images permanently in memory...

File: ReactWindows/ReactNative/Modules/Image/ImageLoaderModule.cs [hunk 3]
```diff
@@ -15,8 +48,382 @@ public override string Name
...
+
+            lock (_imageReferences)
+            {
+                var reference = default(Tuple<int, bool>);
+
+                if (_imageReferences.TryGetValue(uriString, out reference))
+                {
+                    if (reference.Item1 + count == 0)
+                    {
+                        lock (_waitingListImage)
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Image/ImageLoaderModule.cs]: Is there really going to be a lot of contention across all these dictionaries?  Couldn't we instead just use a single lock?

File: ReactWindows/ReactNative/Views/Image/ReactImageManager.cs [hunk 4]
```diff
@@ -188,58 +203,125 @@ protected override Border CreateViewInstance(ThemedReactContext reactContext)
...
+                SetPixels(view, image, tintColor, backgroundColor, pixelData, source);
+            }
+            else
+            {
+                loader.LoadImagePixelData(source, this, Tuple.Create(view, tintColor, backgroundColor));
+            }
         }
 
-        private void OnImageOpened(Border view, RoutedEventArgs args)
+        private async void SetPixels(
```
Comment (rozele) [inline ReactWindows/ReactNative/Views/Image/ReactImageManager.cs]: Can we split this logic for editing images into a helper class?

File: ReactWindows/ReactNative/Modules/Image/ImageLoaderModule.cs [hunk 5]
```diff
@@ -15,8 +48,382 @@ public override string Name
...
+
+            if (image != null)
+            {
+                ImageReference(uriString, 1);
+            }
+
+            return image;
+        }
+
+        internal void FreeReference(string uriString)
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Image/ImageLoaderModule.cs]: Maybe we can take a more object oriented approach and use something like IDisposable.

File: ReactWindows/ReactNative/Modules/Image/ImageLoaderModule.cs [hunk 6]
```diff
@@ -1,9 +1,42 @@
...
+using Windows.Storage.Streams;
+using Windows.UI;
+using Windows.UI.Core;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Media.Imaging;
 
 namespace ReactNative.Modules.Image
 {
     class ImageLoaderModule : NativeModuleBase
```
Comment (rozele) [inline ReactWindows/ReactNative/Modules/Image/ImageLoaderModule.cs]: Does it make sense to prefetch the image to disk first?

**Comment by rozele** [comment]
A general comment is about the architecture.  I think on iOS and Android they use a hybrid in-memory / disk cache.  I imagine we're relatively limited in existing options for this, and any cache work we do would potentially conflict with the existing caching mechanisms built into BitmapImage, but it might make sense to be architecturally consistent.

File: Libraries/Image/Image.windows.js (hunk 1 above)
Comment (ebragge) [inline Libraries/Image/Image.windows.js]: The other missing features relate to: 
1) ShouldNotifyLoadEvents (**added**)
2) RCTTextInlineImage

File: ReactWindows/ReactNative/Views/Image/ReactImageManager.cs (hunk 4 above)
Comment (ebragge) [inline ReactWindows/ReactNative/Views/Image/ReactImageManager.cs]: OK

File: ReactWindows/ReactNative/Views/Image/ImageReference.cs [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+{
+    class ImageReference : IDisposable
+    {
+        private BitmapImage _image;
+        private byte[] _pixelData;
+        private int _referenceCount;
+        private string _key;
+        private Dictionary<string, ImageReference> _imageDictionary;
+
+        internal ImageReference(BitmapImage image, int count, string key, Dictionary<string, ImageReference> dictionary)
```
Comment (rozele) [inline ReactWindows/ReactNative/Views/Image/ImageReference.cs]: The class is internal, so we might as well make all these public.

**Pull request closed without merging**